src/java/org/apache/cassandra/index/sai/disk/vector/OnDiskOrdinalsMap.java

```diff
@@ -231,17 +233,42 @@ private class FileReadingOrdinalsView implements OrdinalsView
     {
         RandomAccessReader reader = fh.createReader();
         private final long high = (segmentEnd - 8 - rowOrdinalOffset) / 8;
+        private final AtomicInteger lastFoundRowId = new AtomicInteger(-1);
+        private final AtomicLong lastFoundRowIdIndex = new AtomicLong(-1);
 
         /**
          * @return order if given row id is found; otherwise return -1
          */
         @Override
         public int getOrdinalForRowId(int rowId) throws IOException
         {
+            long low = 0;
+            if (lastFoundRowId.get() > -1 && lastFoundRowId.get() <= rowId)
+            {
+                low = lastFoundRowIdIndex.get();
+
+                // sequential read, skip binary search
+                if (low + 1 == rowId)
+                {
+                    long offset = rowOrdinalOffset + (lastFoundRowIdIndex.get() + 1) * 8;
+                    reader.seek(offset);
+                    int foundRowId = reader.readInt();
```

Comment:
Do we need to verify there is a next `int` to read? That might happen for rows that are in the sstable but not in the vector index.

Reply:
i'll ad check for "lastFoundRowId < high", like in the binary search